core/shared/src/main/scala-3/monocle/internal/focus/ErrorHandling.scala

```diff
@@ -3,19 +3,52 @@ package monocle.internal.focus
 private[focus] trait ErrorHandling {
   this: FocusBase =>
 
-  def errorMessage(error: FocusError): String = error match {
-    case FocusError.NotACaseClass(fromClass, fieldName) =>
-      s"Cannot generate Lens for field '$fieldName', because '$fromClass' is not a case class"
+  def errorReport(error: FocusError): (String, Option[Position]) = error match {
```

Comment:
The macro has better support for errors now, I need to see whether we can use the `-explain` tag.